src/libstore/unix/build/external-derivation-builder.cc

```diff
@@ -0,0 +1,107 @@
+namespace nix {
+
+struct ExternalDerivationBuilder : DerivationBuilderImpl
+{
+    Settings::ExternalBuilder externalBuilder;
+
+    ExternalDerivationBuilder(
+        Store & store,
+        std::unique_ptr<DerivationBuilderCallbacks> miscMethods,
+        DerivationBuilderParams params,
+        Settings::ExternalBuilder externalBuilder)
+        : DerivationBuilderImpl(store, std::move(miscMethods), std::move(params))
+        , externalBuilder(std::move(externalBuilder))
+    {
+    }
+
+    static std::unique_ptr<ExternalDerivationBuilder> newIfSupported(
+        Store & store, std::unique_ptr<DerivationBuilderCallbacks> & miscMethods, DerivationBuilderParams & params)
+    {
+        for (auto & handler : settings.externalBuilders.get()) {
+            for (auto & system : handler.systems)
+                if (params.drv.platform == system)
+                    return std::make_unique<ExternalDerivationBuilder>(
+                        store, std::move(miscMethods), std::move(params), std::move(handler));
```

Comment:
```suggestion
                        store, std::move(miscMethods), std::move(params), handler);
```

otherwise, this causes the list to be emptied (and attempt to build locally again, which ofc won't work if the system can't build it)